contracts/extensions/ReputationBootstrapper.sol

```diff
@@ -102,6 +99,10 @@ contract ReputationBootstrapper is ColonyExtensionMeta {
 
   // Public
 
+  function setGiveTokens(bool _giveTokens) public onlyRoot {
```

Comment:
Should this be changeable during the process?

Reply:
It's a good point, I worry about going down a recursive sort of rabbit hole. Do we add a "locked" boolean, which prevents more claims from being set? Can the "locked" boolean itself be set back to false? If so, then an admin could go back and disable tokens anyway. We could have the "locked" boolean be one-way, which also checks to make sure there's sufficient tokens available. But someone could always deprecated & uninstall the extension and reclaim the funds, unless we say you cannot delete the extension until all the funds are claimed, but what if then someone loses the email and never comes to claim? Are funds locked permanently?

I would lean in favor of leaving it open, since ultimately a colony is choosing to give tokens here. I think we can assume that they will continue to make that choice. Put another way, I don't think this is such an adversarial setting.

Reply:
As-ever, when it's a question of the actual requirements (rather than the specific implementation), we need to go back to product (-:

Reply:
@arrenv can you weigh in here? Essentially the question is whether the flag to give tokens or not should be locked, or whether we can leave it an open setting which can (in theory) be changed by admins at any point. Locking the flag introduces other considerations, as I've touched on in my earlier comment.

Reply:
From what I'm understanding, the question is whether an admin should be able to toggle the bootstrapping claims. This does not seem very useful unless values could be changed.

It does seem like a good idea to be able to cancel a bootstrapping if a mistake is made, this seems like it could be handled by deprecating the extension.

Reply:
I think a good solution here would be to add a one-way "locked" switch, similar to what we do with expenditures. While in a "draft" state, the parameters can all be changed. Once "locked", the parameters are set and people can begin claiming.

The extension could then be deprecated only once the token balance was down to zero (as then everyone would have claimed, or if the only grants are reputation, there's no cost to removing them).

Would that be acceptable? It gives admins the flexibility to configure the contract, but then gives some guarantees to end users once they start claiming.

Reply:
@kronosapiens Nice, yeah that sounds acceptable to me.